sdk/core/allocator/alloc.h

```diff
@@ -1966,6 +1979,8 @@ class MState
 			unlink_chunk(MChunk::from_header(prev), prev->size_get());
 			ds::linked_list::unsafe_remove_link(prev, p);
 			p->clear();
+			// p is no longer a header. Clear the shadow bit.
+			revoker.shadow_paint_single(CHERI::Capability{p}.address(), false);
```

Comment:
Maybe move the `shadow_paint_single` into `->clear`?

Reply:
Slightly tricky because `clear()` is called on both header creation and destruction. We need to have two wrappers, or pass a boolean to `clear()`?

Reply:
Ah, right, because we might be `clear()`-ing some existing data under where we're `split()`-ing to make a new header.  I'd be tempted to have a private `zero()` called on construction and by the public `clear()`, but it's probably fine as is.